iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/StringLiteralDuplicatedCheck.java

```diff
@@ -0,0 +1,135 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+import org.sonar.check.Rule;
+import org.sonar.check.RuleProperty;
+import org.sonar.iac.arm.tree.api.File;
+import org.sonar.iac.arm.tree.api.FunctionCall;
+import org.sonar.iac.arm.tree.api.ResourceDeclaration;
+import org.sonar.iac.arm.tree.api.StringLiteral;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.common.api.checks.SecondaryLocation;
+import org.sonar.iac.common.api.tree.Tree;
+import org.sonar.iac.common.extension.visitors.TreeContext;
+import org.sonar.iac.common.extension.visitors.TreeVisitor;
+
+@Rule(key = "S1192")
+public class StringLiteralDuplicatedCheck implements IacCheck {
+
+  private static final String MESSAGE = "Define a variable instead of duplicating this literal \"%s\" %d times.";
+  private static final String SECONDARY_MESSAGE = "Duplication.";
+  private static final Pattern ALLOWED_DUPLICATED_LITERALS = Pattern.compile("(?U)^[a-zA-Z_][.\\-\\w]+$");
+
+  public static final int THRESHOLD_DEFAULT = 3;
+  public static final int MINIMAL_LITERAL_LENGTH_DEFAULT = 5;
+
+  @RuleProperty(
+    key = "threshold",
+    defaultValue = "" + THRESHOLD_DEFAULT)
+  protected int threshold = THRESHOLD_DEFAULT;
+
+  @RuleProperty(
+    key = "minimal_literal_length",
+    defaultValue = "" + MINIMAL_LITERAL_LENGTH_DEFAULT)
+  protected int minimalLiteralLength = MINIMAL_LITERAL_LENGTH_DEFAULT;
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(File.class, this::visitFile);
+  }
+
+  private void visitFile(CheckContext ctx, File file) {
+    var stringVisitor = new StringVisitor();
+    file.statements().forEach(stringVisitor::scan);
+    stringVisitor.reportDuplicates(ctx);
+  }
+
+  private class StringVisitor extends TreeVisitor<TreeContext> {
+
+    private static final TreeContext DUMMY_TREE_CONTEXT = new TreeContext();
+    private final Map<String, List<StringLiteral>> sameLiteralOccurrences = new HashMap<>();
+
+    public StringVisitor() {
+      register(StringLiteral.class, (ctx, tree) -> visitLiteral(tree));
+    }
+
+    private void scan(Tree root) {
+      super.scan(DUMMY_TREE_CONTEXT, root);
+    }
+
+    private void visitLiteral(StringLiteral tree) {
+      if (!isIgnored(tree)) {
+        var value = tree.value();
+        if (!sameLiteralOccurrences.containsKey(value)) {
+          List<StringLiteral> occurrences = new ArrayList<>();
+          occurrences.add(tree);
+          sameLiteralOccurrences.put(value, occurrences);
+        } else {
+          sameLiteralOccurrences.get(value).add(tree);
+        }
+      }
+    }
+
+    private boolean isIgnored(StringLiteral stringLiteral) {
+      var value = stringLiteral.value();
+      return value.length() < minimalLiteralLength
+        || ALLOWED_DUPLICATED_LITERALS.matcher(value).find()
```

Comment:
As you're matching the whole string (you're using anchors in the regex), you could use matches() instead.